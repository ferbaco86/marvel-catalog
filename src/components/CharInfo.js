import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import CoverImage from './CoverImage';

const imageSize = 'detail';

const Container = styled.div`
display: flex;
flex-direction: column`;

const DescriptionContainer = styled.div`
margin-top: -8rem;
display: flex;
flex-direction: column;
justify-content: center;
align-items: center;`;

const SubTitle = styled.h3`
color: #202020;
padding: 1rem;
text-transform: uppercase;
font-family: 'Roboto Condensed', sans-serif;
line-spacing: 1px;
font-size: 2rem;
text-align: center;`;

const Description = styled.p`
font-size: 1.2rem;
color: #202020;
padding: 3rem;
text-align: center;`;

const MoreInfoLink = styled.a`
font-size: 1.3rem;
text-decoration: underline;
color: #ec1d24`;

const CharInfo = props => {
  const {
    name, image, extension, description, url,
  } = props;

  let descriptionText = '';

  if (description === '') {
    descriptionText = 'No description provided';
  } else {
    descriptionText = description;
  }
  return (
    <Container>
      <CoverImage name={name} imageURL={`${image}/${imageSize}.${extension}`} />
      <DescriptionContainer>
        <SubTitle>Description</SubTitle>
        <Description>{descriptionText}</Description>
        <MoreInfoLink href={url} target="_blank">More Info</MoreInfoLink>
      </DescriptionContainer>
    </Container>
  );
};

CharInfo.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  extension: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default CharInfo;
