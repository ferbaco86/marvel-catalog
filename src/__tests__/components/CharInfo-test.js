import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/index';
import CharInfo from '../../components/CharInfo';

const character = {
  name: 'Spider-Man',
  description: 'Super Hero',
  image: 'www.mockurl.com',
  extension: 'jpg',
  url: 'www.mockurl2.com',
};

describe('Rendering component', () => {
  it('creates a CharInfo component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <CharInfo
              name={character.name}
              description={character.description}
              image={character.image}
              extension={character.extension}
              url={character.url}
            />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('Spider-Man');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <CharInfo
            name={character.name}
            description={character.description}
            image={character.image}
            extension={character.extension}
            url={character.url}
          />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
