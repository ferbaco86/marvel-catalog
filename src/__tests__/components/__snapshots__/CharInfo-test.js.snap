// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div
  className="sc-eCssSg gxNdKZ"
>
  <div
    className="sc-bdfBwQ hNRTSc"
  >
    <div
      alt="Spider-Man Cover"
      className="sc-gsTCUz fMlBTs"
    >
      <h1
        className="sc-dlfnbm AgPsk"
      >
        Spider-Man
      </h1>
      <h3
        className="sc-hKgILt hOQUZQ"
      >
        
      </h3>
    </div>
  </div>
  <div
    className="sc-jSgupP brJnBv"
  >
    <h3
      className="sc-gKsewC jkmbiG"
    >
      Description
    </h3>
    <p
      className="sc-iBPRYJ cgYlZU"
    >
      Super Hero
    </p>
    <a
      className="sc-fubCfw dbIUh"
      href="www.mockurl2.com"
      target="_blank"
    >
      More Info
    </a>
  </div>
</div>
`;
